sign/signup/controller.js

```diff
@@ -0,0 +1,185 @@
+import { ERROR_MESSAGES, EMAIL_REGEX, PASSWORD_REGEX, API } from '../constants.js';
+
+const getErrorMessage = (errCode) => ERROR_MESSAGES[errCode] ?? '알 수 없는 에러가 발생했습니다.';
+const getAPI = (apiCode) => API[apiCode];
+
+export default class SignUpController {
+  constructor(model, view) {
+    this.model = model;
+    this.view = view;
+
+    this.validateUser();
+    this.view.emailInput.addEventListener('focusout', () => this.handleEmailInputFocusout());
+    this.view.emailInput.addEventListener('keydown', () => this.handleEmailInputKeydown());
+    this.view.passwordInput.addEventListener('focusout', () => this.handlePasswordInputFocusout());
+    this.view.passwordInput.addEventListener('keydown', () => this.handlePasswordInputKeydown());
+    this.view.passwordRepeatInput.addEventListener('focusout', () =>
+      this.handlePasswordRepeatInputFocusout(),
+    );
+    this.view.passwordRepeatInput.addEventListener('keydown', () =>
+      this.handlePasswordRepeatInputKeydown(),
+    );
+    this.view.eyeIcons.forEach((button) => {
+      button.addEventListener('click', (e) => this.view.handleVisibilityIconClick(e));
+    });
+    this.view.formElement.addEventListener('submit', (e) => this.handleFormSubmit(e));
```

Comment:
화살표 함수 대신 bind를 사용해서 이렇게 하실 수도 있습니다.
[MDN - bind](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) 참고해보세요. 😄 
```suggestion
    this.view.formElement.addEventListener('submit',  this.handleFormSubmit.bind(this));
```

Reply:
오 bind를 사용해서도 this를 지정할 수 있군요!! call만 최근에 알았었는데 새로운 메소드를 알았네요 감사합니다!!